packages/atomic/src/components/search/atomic-search-box-recent-queries/atomic-search-box-recent-queries.spec.ts

```diff
@@ -0,0 +1,166 @@
+import {html} from 'lit';
+import {beforeEach, describe, expect, it, vi} from 'vitest';
+import {fixture} from '@/vitest-utils/testing-helpers/fixture';
+import type {AtomicSearchBoxRecentQueries} from './atomic-search-box-recent-queries';
+import './atomic-search-box-recent-queries';
+
+vi.mock('@coveo/headless', {spy: true});
+
+vi.mock('../../../utils/local-storage-utils', () => ({
+  SafeStorage: vi.fn().mockImplementation(() => ({
+    getParsedJSON: vi.fn().mockReturnValue(['query1', 'query2', 'query3']),
+    setJSON: vi.fn(),
+    removeItem: vi.fn(),
+  })),
+  StorageItems: {
+    RECENT_QUERIES: 'recent-queries',
+  },
+}));
+
+vi.mock('../../../utils/utils', () => ({
+  once: (fn: () => void) => fn,
+}));
+
+vi.mock('../../common/suggestions/suggestions-events', () => ({
+  dispatchSearchBoxSuggestionsEvent: vi.fn(),
+}));
+
+describe('AtomicSearchBoxRecentQueries', () => {
+  const renderComponent = async (
+    props: {icon?: string; maxWithQuery?: number; maxWithoutQuery?: number} = {}
+  ) => {
+    const element = await fixture<AtomicSearchBoxRecentQueries>(
+      html`<atomic-search-box-recent-queries></atomic-search-box-recent-queries>`
+    );
+
+    if (props.icon !== undefined) element.icon = props.icon;
+    if (props.maxWithQuery !== undefined)
+      element.maxWithQuery = props.maxWithQuery;
+    if (props.maxWithoutQuery !== undefined)
+      element.maxWithoutQuery = props.maxWithoutQuery;
+
+    await element.updateComplete;
+
+    return element;
+  };
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('should render correctly', async () => {
```

Comment:
These tests are quite different from those in the atomic-commerce-search-box-recent-queries spec file!

I suggest you rather start from this spec file, and make adjustments to cover any behaviors that are specific to atomic search, and possibly improve both specs if you find additional common behaviors that should be tested.